packages/definitions-parser/src/get-affected-packages.ts

```diff
@@ -1,109 +1,82 @@
+import { AllPackages, PackageId, formatTypingVersion, getDependencyFromFile } from "./packages";
+import { resolve } from "path";
+import { satisfies } from "semver";
+export interface PreparePackagesResult {
+  readonly packageNames: Set<string>;
+  readonly dependents: Set<string>;
 }
 
 /** Gets all packages that have changed on this branch, plus all packages affected by the change. */
-export function getAffectedPackages(allPackages: AllPackages, changedPackageIds: PackageId[]): Affected {
-  const resolved = changedPackageIds.map((id) => allPackages.tryResolve(id));
-  // If a package doesn't exist, that's because it was deleted.
-  const changed = mapDefined(resolved, (id) => allPackages.tryGetTypingsData(id));
-  const dependent = mapIterable(collectDependers(resolved, getReverseDependencies(allPackages, resolved)), (p) =>
-    allPackages.getTypingsData(p)
-  );
-  return { changedPackages: changed, dependentPackages: sortPackages(dependent), allPackages };
-}
-
-/** Every package name in the original list, plus their dependencies (incl. dependencies' dependencies). */
-export function allDependencies(allPackages: AllPackages, packages: Iterable<TypingsData>): TypingsData[] {
-  return sortPackages(transitiveClosure(packages, (pkg) => allPackages.allDependencyTypings(pkg)));
-}
-
-/** Collect all packages that depend on changed packages, and all that depend on those, etc. */
-function collectDependers(
-  changedPackages: PackageId[],
-  reverseDependencies: Map<PackageId, Set<PackageId>>
-): Set<PackageId> {
-  const dependers = transitiveClosure(changedPackages, (pkg) => reverseDependencies.get(pkg) || []);
-  // Don't include the original changed packages, just their dependers
-  for (const original of changedPackages) {
-    dependers.delete(original);
-  }
-  return dependers;
-}
-
-function sortPackages(packages: Iterable<TypingsData>): TypingsData[] {
-  return sort<TypingsData>(packages, PackageBase.compare); // tslint:disable-line no-unbound-method
-}
-
-function transitiveClosure<T>(initialItems: Iterable<T>, getRelatedItems: (item: T) => Iterable<T>): Set<T> {
-  const all = new Set<T>();
-  const workList: T[] = [];
-
-  function add(item: T): void {
-    if (!all.has(item)) {
-      all.add(item);
-      workList.push(item);
+export async function getAffectedPackages(
+  allPackages: AllPackages,
+  deletions: PackageId[],
+  definitelyTypedPath: string
+): Promise<PreparePackagesResult> {
+  const allDependents = [];
+  const filters = [`--filter '...[${sourceRemote}/${sourceBranch}]'`];
+  for (const d of deletions) {
+    for (const dep of allPackages.allTypings()) {
+      for (const [name, version] of dep.allPackageJsonDependencies()) {
+        if (
+          "@types/" + d.typesDirectoryName === name &&
+          (d.version === "*" || satisfies(formatTypingVersion(d.version), version))
+        ) {
+          filters.push(`--filter '...{./types/${dep.name}}'`);
```

Comment:
I think it will: dep.name should be, eg, react which I *think* will include react/v15 if v15 depends on the deleted package.

And if you delete an old version, then on the previous line with `satisfies`, both deletions and dep.allPackageJsonDependencies will correctly produce whatever/v2 for `d.version` and `version`, respectively.

Reply:
But, I thought after Andrew's change, `dep.name` is now `@types/react`? I would expect this to be `dep.typesDirectoryName` as that's the path being used here.

Reply:
Or just `` `...${dep.name}` ``, right?

Reply:
Probably that, since this doesn't seem to include versioning, may as well install the whole package even if we have more than one. Likely safer.